napari/_qt/qt_event_loop.py

```diff
@@ -132,6 +133,17 @@ def get_app(
         app.setWindowIcon(QIcon(kwargs.get('icon')))
         set_app_id(kwargs.get('app_id'))
 
+        if os.getenv("NAPARI_CATCH_ERRORS") not in ('0', 'False'):
+
+            app.arena = []
+
+            def show_notification(*args, **kwargs):
```

Comment:
I assume this is here as a solution to your garbage collection issues... but I'd much prefer this be a static method on NapariQtNotification itself, rather than a dynamic function defined here.  For one thing, it would make testing this _much_ easier as we can just `mock.patch` `NapariQtNotification.show_notification` and make sure it gets called with the expected kwargs.

But also, it just feels like it belongs there, and not here in `qt_event_loop.py`.

Reply:
I can't make it a purely a static method, I need access to an object to keep track of the notification instance; I can make it a class method and attach to the class if you want.

Reply:
that's fine, make it a class method

Reply:
but also... I'm still unsatisfied with our explanation and reproducibility of the garbage collection issue.  It's not that I don't think it's potentially an issue to not save the issue, but I'd like us to understand _why_ it's an issue, before just saving references as a "patch".

Reply:
Ok, I can't trigger it anymore, 

It used to be that
```
    @classmethod
    def show_notification(cls, notification):
        cls.from_notification(notification).show()
        import gc
        gc.collect()
```

Would crash napari for me.

Reply:
i thought it was crashing for you only when you clicked the show traceback button (creating a second window that had no reference)?

Reply:
Forcing gc collection was also triggering the issue; that's how I tracked down the initial viewer missing references. 
the showing traceback was relatively good at triggering those issue, but it was not the only thing; plus like any other GC stuff, it would often move as soon as some code was commented or added.